homework_basic_components/src/task8/Task8.java

```diff
@@ -0,0 +1,166 @@
+package task8;
+
+import java.awt.BorderLayout;
+import java.awt.GridLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JPasswordField;
+import javax.swing.JTextField;
+
+/**
+ * Represents small GUI application for registration.
+ * Registration will be successful if username and password 
+ * are longer than 6 characters, and if password contains at least
+ * one special character and one number.
+ * @author ajla
+ *
+ */
+public class Task8 extends JFrame {
```

Comment:
If you provide password `jbegic%5` it will not correctly identify the number and will display invalid error message saying that number is required.